lib/main.py

```diff
@@ -27,57 +27,54 @@ async def lifespan(_app: FastAPI):
 
 
 APP = FastAPI(lifespan=lifespan)
-TASK_LIST: queue.Queue = queue.Queue(maxsize=100)
 
 
 class BackgroundProcessTask(threading.Thread):
     def run(self, *args, **kwargs):  # pylint: disable=unused-argument
+        nc = NextcloudApp()
+
+        provider_ids = set()
+        task_type_ids = set()
+        for chain_name, _ in chains.items():
+            provider_ids.add("llm2:" + chain_name)
+            (model, task) = chain_name.split(":", 2)
+            task_type_ids.add("core:text2text:" + task)
+
         while True:
-            task = TASK_LIST.get(block=True)
+            response = nc.providers.task_processing.next_task(list(provider_ids), list(task_type_ids))
+            if not isinstance(response, dict):
```

Comment:
here we should check for empty dict
```python3
if not response:
   time.sleep(5)
```

Reply:
It's either an empty list or a dict with the correct values. So we need to check if the response is a dict.

Reply:
Ah you changed this on the server side...